src/datanode/src/sql/alter.rs

```diff
@@ -13,18 +13,25 @@
 // limitations under the License.
 
 use catalog::RenameTableRequest;
+use common_procedure::{watcher, ProcedureManagerRef, ProcedureWithId};
 use common_query::Output;
+use common_telemetry::logging::info;
 use snafu::prelude::*;
 use sql::statements::alter::{AlterTable, AlterTableOperation};
 use sql::statements::column_def_to_schema;
 use table::engine::{EngineContext, TableReference};
 use table::requests::{AddColumnRequest, AlterKind, AlterTableRequest};
+use table_procedure::AlterTableProcedure;
 
 use crate::error::{self, Result};
 use crate::sql::SqlHandler;
 
 impl SqlHandler {
     pub(crate) async fn alter(&self, req: AlterTableRequest) -> Result<Output> {
+        if let Some(procedure_manager) = &self.procedure_manager {
```

Comment:
Is this the final form（code style） in which `procedure` is used?

```
if has procedure {
    do something with procedure
} else {
    do the same thing without procedure
}
```

Reply:
Maybe we can always use procedures to do this when procedures for all table operations are ready. Then we don't need an `if-else` check.

Reply:
A TODO here?

Reply:
There is a related todo in SqlHandler, we need to support multiple table engines and then we can enable the procedure framework by default.